### ISSUE TITLE
Cannot iterate over docker tags in repo. Is my database corrupt?

### ISSUE THREAD

**Issue by lukeplausin** [goharbor/harbor-helm#504]
Hi, I'm experiencing some problems after an upgrade. I've searched around online but couldn't find any reports of similar issues.

I am using S3 storage and an external Postgres database. For other reasons it was necessary to delete the entire kubernetes master before redeploying harbor. During the initial helm install, the deployment initially failed because the S3 access key was set incorrectly.

Now I have harbor up and running. I can log into the portal and see the projects and repositories, but when I click on the repo I cannot see the image tags. In the UI a popup appears reading `Internal Server Error`. In the logs I note these corresponding events:

* Core *
```
2020-02-06T18:39:57Z [ERROR] [/common/api/base.go:234]: failed to get tag of redacted/redacted: http error: code 401, message {"errors":[{"code":"UNAUTHORIZED","message":"authentication required","detail":[{"Type":"repository","Class":"","Name":"redacted/redacted","Action":"pull"}]}]}
2020-02-06T18:39:57Z [ERROR] [/common/api/base.go:74]: GET /api/repositories/redacted/redacted/tags?detail=true failed with error: {"code":500,"message":"Internal Server Error"}
2020/02/06 18:39:57.354 [D] [middleware.go:41]  |151.237.232.133| 500 |   17.06345ms|   match| GET      /api/repositories/redacted/redacted/tags   r:/api/repositories/*/tags
```

* Registry *
```
10.10.156.217 - - [06/Feb/2020:18:39:57 +0000] "GET /v2/redacted/redacted/tags/list HTTP/1.1" 401 162 "" "harbor-registry-client"
time="2020-02-06T18:39:57.354032095Z" level=warning msg="error authorizing context: authorization token required" go.version=go1.12.12 http.request.host="redacted-harbor-prod-harbor-registry:5000" http.request.id=d7fdcc54-012e-4eeb-a5d3-4ce183b12560 http.request.method=GET http.request.remoteaddr="10.10.156.217:39720" http.request.uri="/v2/redacted/redacted/tags/list" http.request.useragent=harbor-registry-client vars.name="redacted/redacted"
```

I've checked the database and S3 credentials and they all seem to be fine. Any ideas what could be causing these issues?

Would it help if I run the registry container with `SYNC_REGISTRY` set?

**Comment by lukeplausin** [comment]
On further inspection it looks like two different issues.

1. The `tags` table in the `registry` postgres database is empty. I guess I will have to run with `SYNC_REGISTRY` to repopulate it.
2. The registry service returns a HTTP 401 error whenever the `harbor-registry-client` tries to connect to it. From what I understand the registry service is doing this because it requires an Authorization header but Harbor core isn't supplying one. Strange.

I will keep investigating.

**Comment by lukeplausin** [comment]
Looks like this was a database issue caused by unstable code running on an image with the `dev` branch.

In case anyone else has this issue - I fixed my environment by deleting and repopulating the database, and copying the projects and repos data from a backup. Harbor retrieved the images from durable storage and no data was lost.

When I set up my environment following the steps in `README.md`, it didn't say anywhere that the helm charts in `master` are considered unstable. Maybe it would be a good idea to add `--version 1.3.0` in the README.md example?
